src/sources/exec.rs

```diff
@@ -0,0 +1,843 @@
+            assert_eq!(2, buffer.len());
+            assert_eq!(vec![0xF0, 0x9F], buffer);
+        } else {
+            panic!("The buffer should be converted to a string");
+        }
+    }
+
+    #[test]
+    fn test_buffer_to_string_invalid_utf8() {
+        let mut buffer: Vec<u8> = vec![
+            0x46, 0x61, 0x73, 0x74, 0x20, 0xF0, 0x9F, 0x9A, 0x80, 0xF0, 0x9F,
+        ];
+
+        if buffer_to_string(&mut buffer, false).is_some() {
+            panic!("The buffer should be not converted to a string");
+        } else {
+            assert_eq!(11, buffer.len());
+        }
+    }
+
+    #[tokio::test]
+    async fn test_spawn_reader_thread_per_line() {
+        trace_init();
+
+        let buf = Cursor::new("hello world\nhello rocket 🚀");
+        let reader = BufReader::new(buf);
+        let shutdown = ShutdownSignal::noop();
+        let (sender, mut receiver) = channel(1024);
+
+        spawn_reader_thread(reader, shutdown, true, 88888, STDOUT, sender);
+
+        let mut counter = 0;
+        if let Some((line, stream)) = receiver.recv().await {
+            assert_eq!("hello world", line);
+            assert_eq!(STDOUT, stream);
+            counter += 1;
+        }
+
+        if let Some((line, stream)) = receiver.recv().await {
+            assert_eq!("hello rocket 🚀", line);
+            assert_eq!(STDOUT, stream);
+            counter += 1;
+        }
+
+        assert_eq!(counter, 2);
+    }
+
+    #[tokio::test]
+    async fn test_spawn_reader_thread_per_blob() {
+        trace_init();
+
+        let buf = Cursor::new("hello world\nhello rocket 🚀");
+        let reader = BufReader::new(buf);
+        let shutdown = ShutdownSignal::noop();
+        let (sender, mut receiver) = channel(1024);
+
+        spawn_reader_thread(reader, shutdown, false, 88888, STDOUT, sender);
+
+        let mut counter = 0;
+        if let Some((line, stream)) = receiver.recv().await {
+            assert_eq!("hello world\nhello rocket 🚀", line);
+            assert_eq!(STDOUT, stream);
+            counter += 1;
+        }
+
+        assert_eq!(counter, 1);
+    }
+
+    #[tokio::test]
+    #[cfg(not(target_os = "windows"))]
+    async fn test_run_command_linux() {
+        trace_init();
+        let config = standard_scheduled_test_config();
+        let (tx, mut rx) = Pipeline::new_test();
+        let shutdown = ShutdownSignal::noop();
+
+        // Wait for our task to finish, wrapping it in a timeout
+        let timeout = tokio::time::timeout(
+            time::Duration::from_secs(5),
+            run_command(config.clone(), shutdown, tx),
+        );
+
+        let timeout_result = timeout.await;
+
+        match timeout_result {
+            Ok(output) => match output {
+                Ok(exit_status) => assert_eq!(0_i32, exit_status.unwrap().code().unwrap()),
+                Err(_) => panic!("Unable to run linux command"),
+            },
+            Err(_) => panic!("Timed out during test of run linux command."),
+        }
```

Comment:
Nit: you could simplify this a bit like:

```rust
let output = timeout.expect("command timed out").expect("command errored")
```

Reply:
Fixed